src/secrets.rs

```diff
@@ -1,22 +1,143 @@
 //! Types for working with registry access secrets
 
+use serde::Deserialize;
+
+use base64;
+use dirs;
+use std::collections::HashMap;
+use std::env;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+use std::process::{Command, Stdio};
+use std::str;
+use tracing::debug;
+
 /// A method for authenticating to a registry
 pub enum RegistryAuth {
     /// Access the registry anonymously
     Anonymous,
+
     /// Access the registry using HTTP Basic authentication
     Basic(String, String),
+
+    /// Access the registry using credentials configured in the Docker config file
+    Default,
 }
 
 pub(crate) trait Authenticable {
-    fn apply_authentication(self, auth: &RegistryAuth) -> Self;
+    fn apply_authentication(self, registry: &str, auth: &RegistryAuth) -> Self;
```

Comment:
Instead of passing a registry, I'd recommend making `RegistryAuth::Default` be `RegistryAuth::Default(String)`. This means no API change and also makes sure the config information is part of the enum